### PR TITLE
AS-908 Add pass through API for ToS text from Orch to Sam

### PR DESCRIPTION
**Pull request by jordansafer**
Added an API at /tos/text for getting the ToS text from SAM as part of the Terms of Service architecture changes being worked on. https://broadworkbench.atlassian.net/browse/AS-908

Testing:
- Unit tested, locally tested, and added an integration test

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [x] Get two thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala [hunk 1]
```diff
@@ -54,5 +60,17 @@ class RegistrationApiSpec extends FreeSpec with Matchers with ScalaFutures with
...
+
+    "should return terms of services with no auth token" in {
+      val req = HttpRequest(GET, ServiceTestConfig.FireCloud.orchApiUrl + s"tos/text")
+      val response = sendRequest(req)
+
+      val textFuture = Unmarshal(response.entity).to[String]
+
+      response.status shouldEqual StatusCodes.OK
+      whenReady(textFuture) { text =>
+        text should include("Terms as of February 12, 2020.")
```
Comment (MatthewBemis) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala]: I am worried that a change to the ToS will break this test. Is there something else that we could assert that would be resilient to ToS changes?

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/RegisterApiService.scala [hunk 2]
```diff
@@ -43,4 +43,14 @@ trait RegisterApiService extends Directives with RequestBuilding with StandardUs
         }
       }
     }
+
+  val tosRoutes: Route =
+    pathPrefix("tos") {
+      path("text") {
+        get {
+          complete {
+            registerServiceConstructor().getTermsOfServiceText }
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/RegisterApiService.scala]: could this use the `passthrough` directive instead? Then you wouldn't need to add anything to `SamDAO`

**Review by sagehen03 — APPROVED** [review]
Works for me!

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala (hunk 1 above)
Comment (jordansafer) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala]: This makes sense. Truthfully, the only thing we know for sure is that the response status is OK. The assertion about the text contents is could always flake. The same issue is present in that SAM integration tests I added, which I can update-

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/RegisterApiService.scala (hunk 2 above)
Comment (jordansafer) [reply src/main/scala/org/broadinstitute/dsde/firecloud/webservice/RegisterApiService.scala]: Oh cool! Testing this out-

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala (hunk 1 above)
Comment (MatthewBemis) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala]: Could it test that the text is not empty? That seems like a safe thing for us to assume in the tests and it gives _some_ information

Comment (jordansafer) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/orch/RegistrationApiSpec.scala]: Yes, it can, I'll test that

**Review by MatthewBemis — APPROVED** [review]
LGTM!

**Pull request merged**